### PR TITLE
Only include Silo header if MPI enabled

### PR DESCRIPTION
**Pull request by streeve**
Add CI build without MPI

### CODE REVIEW

**Comment by junghans** [comment]
I think we should create a container without mpi for that! I found cases before where we include mpi.h unconditionally!

**Comment by streeve** [comment]
Makes sense - reopening only to make sure we get this fix in relatively soon

**Comment by junghans** [comment]
> I think we should create a container without mpi for that! I found cases before where we include mpi.h unconditionally!

See https://github.com/ECP-copa/ci-containers/pull/15

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -165,6 +165,17 @@ jobs:
             heffte: 'NoheFFTe'
             hypre: 'NoHYPRE'
             coverage: 'OFF'
+          - distro: 'ubuntu:latest'
```
Comment (junghans) [inline .github/workflows/CI.yml:168]: With https://github.com/ECP-copa/ci-containers/pull/15 merged, we could use
```suggestion
          - distro: 'fedora:nompi'
```
here

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -365,7 +380,7 @@ jobs:
             -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }} \
             -DVALGRIND_EXECUTABLE=False \
             -DDOXYGEN_WARN_AS_ERROR=FAIL_ON_WARNINGS \
-            ${cabana_cmake_opts}
+            ${cabana_cmake_opts[@]}
```
Comment (junghans) [inline .github/workflows/CI.yml:383]: Good catch!

**Comment by streeve** [comment]
@junghans this should be ready

**Pull request merged**